flycheck.el

```diff
@@ -6275,11 +6275,30 @@ See URL `http://www.haskell.org/ghc/'."
   :modes (haskell-mode literate-haskell-mode)
   :next-checkers ((warning . haskell-hlint)))
 
+(flycheck-def-config-file-var flycheck-hlintrc haskell-hlint "HLint.hs"
+  :safe #'stringp)
+
+(flycheck-def-option-var flycheck-hlint-options
+    '("-XQuasiQuotes" "-XTemplateHaskell") haskell-hlint
```

Comment:
Please remove the default values.  Flycheck should follow hlint's defaults, and if hlint doesn't enable these extensions by default, Flycheck should neither.